spark/src/main/java/com/netflix/iceberg/spark/source/IcebergSource.java

```diff
@@ -89,10 +90,10 @@ public DataSourceReader createReader(DataSourceOptions options) {
           .toUpperCase(Locale.ENGLISH));
     }
 
-    return Optional.of(new Writer(table, lazyConf(), format));
+    return Optional.of(new Writer(table, format, table.operations().io()));
   }
 
-  protected Table findTable(DataSourceOptions options) {
+  protected TableWithTableOperations findTable(DataSourceOptions options) {
```

Comment:
@rdblue this is the most interesting question raised by this patch. What do you think?

Reply:
What about adding `io` to the `Table` interface? I'd rather do that since `FileIO` is a public interface. I think that is mostly what HasTableOperations is used for anyway.

Reply:
I think that's fine.